src/main/java/side/util/Storage.java

```diff
@@ -0,0 +1,119 @@
+package side.util;
+
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import side.tasks.Deadline;
+import side.tasks.Event;
+import side.tasks.Task;
+
+/**
+ * Models saving to hard disk using a .txt file. It contains logic for saving and retrieving from
+ * the .txt file.
+ *
+ * @author Lua Yi Da
+ */
+
+public class Storage {
+
+    private Parser parser;
+
+    /**
+     * Initialises new Storage instance with a parser.
+     */
+    public Storage() {
+        this.parser = new Parser();
+    }
+
+    /**
+     * Helper method to save existing tasks to .txt file.
+     *
+     * @param savedList ArrayList containing existing tasks.
+     * @param directory String representing directory to save to.
+     * @throws IOException Error with file operation.
+     */
+    public void save(ArrayList<Task> savedList, String directory) throws IOException {
+        String filePath = directory + "/Side.txt";
+        File file = new File(filePath);
+        File dataDirectory = new File(directory);
+
+        if (!Files.exists(Path.of(directory))) {
+            boolean isDirectory = dataDirectory.mkdir();
+        }
+        if (!Files.exists(Path.of(filePath))) {
+            boolean isFile = file.createNewFile();
+        }
+
+        FileWriter writer = new FileWriter(filePath);
+        StringBuilder taskLine = new StringBuilder();
+
+        for (Task t : savedList) {
+            if (t instanceof Deadline) {
+                String taskDetails = "D | " + ((Deadline) t).getDatetime().saveDatetime();
+                taskLine.append(taskDetails);
+            } else if (t instanceof Event) {
+                String taskDetails = "E | " + ((Event) t).getStartDatetime().saveDatetime()
+                        + "/to" + ((Event) t).getEndDatetime().saveDatetime();
+                taskLine.append(taskDetails);
+            } else {
+                taskLine.append("T | No time");
+            }
```

Comment:
Maybe you can apply polymorphism in this part as well, just like how you did for the printed output.